coverage-experiment/src/main/java/nl/tudelft/serg/evosql/experiment/Runner.java

```diff
@@ -1,18 +1,135 @@
+        Stream<String> suitecrm = reader_suitecrm.lines();
+
+
+        QueryReader queryReader = new QueryReader();
+        List<String> allQueries = queryReader.readQueries(erpnext, espocrm, suitecrm);
+
+        // TODO: Make connection data for separate databases...
+        for (int i = startIndex; i < allQueries.size(); i += stepSize) {
+            runForQuery(
+                    allQueries.get(i),
+                    null,
+                    null,
+                    String.valueOf(i)
+            );
+        }
+
+        try {
+            reader_erpnext.close();
+            reader_espocrm.close();
+            reader_suitecrm.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
     }
+
+    /**
+     * End to end execution of experiment for a single query.
+     *
+     * @param query              query to execute
+     * @param connectionDataProd data of db connection for production database
+     * @param connectionDataTest data of db connection for test database
+     */
+    public static QueryExperimentResult runForQuery(String query,
+                                             ConnectionData connectionDataProd,
+                                             ConnectionData connectionDataTest,
+                                             String queryClassName) {
+        // TODO: Get file package
+        BrewExecutor brewExecutor = new BrewExecutor(connectionDataProd, connectionDataTest, "");
+        // TODO: Pass a good folder name which is useable
+        brewExecutor.executeBrew(query, Paths.get("../test"), queryClassName + "_original.java");
+
+        //String mutatedQuery = QueryMutator.mutateQuery(query);
+        brewExecutor.brewWithMutatedQuery(query, brewExecutor.getQueryResult(), Paths.get("../test"), queryClassName + "_mutated.java");
+
+        // TODO: Get right path
+        org.junit.runner.Result[] results = testClassRunner(
+                Paths.get("../test").toAbsolutePath().toString(),
+                queryClassName + "_original.java",
+                queryClassName + "_mutated.java"
+        );
+
+        QueryExperimentResult experimentResult = new QueryExperimentResult(
+                query,
+                "",
+                results[0].wasSuccessful(),
+                results[1].wasSuccessful()
+        );
+
+        return experimentResult;
+    }
+
+
+    /**
+     * Runs generated test classes and returns a jUnit result datastructure.
+     *
+     * @param folderPath             absolute path containing the test classes to run
+     * @param originalQueryClassName test class name of the 'correct' query
+     * @param mutatedQueryClassName  test class name of the mutated query
+     * @return test results, first index is the original query result, second index is the mutated query result.
+     */
+    public static org.junit.runner.Result[] testClassRunner(String folderPath,
```

Comment:
If we are to use Gradle for running the tests, this one becomes unnecessary as well. In fact, the JUnit 4 dependency can then be dropped from the project.

Reply:
I removed the entire method